agent-inject/agent/config.go

```diff
@@ -95,9 +96,16 @@ func (a *Agent) newTemplateConfigs() []*Template {
 			template = fmt.Sprintf(DefaultTemplate, secret.Path)
 		}
 
+		filePathAndName := secret.FilePathAndName
+		if filePathAndName == "" {
+			filePathAndName = fmt.Sprintf("%s/%s", secret.MountPath, secret.Name)
+		} else {
+			filePathAndName = filepath.Join(secret.MountPath, filePathAndName)
+		}
```

Comment:
I think `filePathAndName := secret.FilePathAndName` ends up being ineffective because in either condition it immediately gets overwritten. I suggest refactoring to something like so:

```go
filePathAndName := fmt.Sprintf("%s/%s", secret.MountPath, secret.Name)

if secret.FilePathAndName != "" {
	filePathAndName = filepath.Join(secret.MountPath, secret.FilePathAndName)
}
```

That should be equivalent to what we have here, and I think with just a single conditional (no `else`) it's easier to reason about: https://play.golang.org/p/-t3Skfz9Yf_k

Reply:
Thanks! This is much cleaner. Added in b96d6b4.